src/server/db_slice.cc

```diff
@@ -787,6 +794,14 @@ pair<PrimeIterator, ExpireIterator> DbSlice::ExpireIfNeeded(const Context& cntx,
   if (time_t(cntx.time_now_ms) < expire_time)
     return make_pair(it, expire_it);
 
+  // Replicate any expiration as DEL command.
+  // TODO: Pass optional key to skip decoding.
+  if (auto journal = owner_->journal(); journal) {
+    string scratch;
+    auto payload = make_pair("DEL"sv, ArgSlice{it->first.GetSlice(&scratch)});
+    journal->RecordEntry(0, journal::Op::COMMAND, cntx.db_index, 1, payload);
```

Comment:
Do you know to tell if this flow will not happen while executing multi command?
If it does, it will break the current impl in replica as if it gets multi command from journal it reads more multi commands till it gets exec. Replica assumes that there is no command op entry that was inserted before the exec op was reached